lib/task/tools/deleteDescriptionTask.class.php

```diff
@@ -49,44 +53,115 @@ protected function configure()
    */
   public function execute($arguments = array(), $options = array())
   {
-    $databaseManager = new sfDatabaseManager($this->configuration);
-    $conn = $databaseManager->getDatabase('propel')->getConnection();
-    sfContext::createInstance($this->configuration);
+    parent::execute($arguments, $options);
+
+    $this->resourceType = $options['repository'] ? 'QubitRepository' : 'QubitInformationObject';
+    $this->fetchResource($arguments['slug']);
+
+    if (!$this->confirmDeletion($options['no-confirmation']))
+    {
+      $this->logSection(sprintf('[%s] Task aborted.', strftime('%r')));
+      return;
+    }
+
+    // User wishes to proceed, delete targeted information objects.
+    switch ($this->resourceType)
+    {
+      case 'QubitRepository':
+        $this->deleteDescriptionsFromRepository();
+        break;
+      case 'QubitInformationObject':
+        $this->deleteDescriptions($this->resource);
```

Comment:
Actually, I remember why I did it now. I also call deleteDescriptions but from inside deleteDescriptionsFromRepository, when it needs to delete something other than $this->resource.